### PR TITLE
Hand-specific HandRenderEvent

### PR DESCRIPTION
**Pull request by ChunHin**
#2747 Update HandRenderEvent so that it is hand-specific
#2811 Re-targeted to master

### CODE REVIEW

**Comment by AEnterprise** [comment]
has merge conflicts

**Comment by ChunHin** [comment]
Solve merge conflicts

File: src/main/java/net/minecraftforge/common/ForgeHooks.java [hunk 1]
```diff
@@ -561,9 +561,9 @@ public static ITextComponent onServerChatEvent(NetHandlerPlayServer net, String
 
 
     static final Pattern URL_PATTERN = Pattern.compile(
-            //         schema                          ipv4            OR           namespace                 port     path         ends
-            //   |-----------------|        |-------------------------|  |----------------------------|    |---------| |--|   |---------------|
-            "((?:[a-z0-9]{2,}:\\/\\/)?(?:(?:[0-9]{1,3}\\.){3}[0-9]{1,3}|(?:[-\\w_\\.]{1,}\\.[a-z]{2,}?))(?::[0-9]{1,5})?.*?(?=[!\"\u00A7 \n]|$))",
+            //         schema                          ipv4            OR        namespace                 port     path         ends
+            //   |-----------------|        |-------------------------|  |-------------------------|    |---------| |--|   |---------------|
+            "((?:[a-z0-9]{2,}:\\/\\/)?(?:(?:[0-9]{1,3}\\.){3}[0-9]{1,3}|(?:[-\\w_]{1,}\\.[a-z]{2,}?))(?::[0-9]{1,5})?.*?(?=[!\"\u00A7 \n]|$))",
```
Comment (RainWarrior) [inline src/main/java/net/minecraftforge/common/ForgeHooks.java]: update before generating patches

**Comment by ChunHin** [comment]
Updated

**Comment by ChunHin** [comment]
Uhoh...

**Comment by ChunHin** [comment]
I wonder why a simple pull request can take sooooo... longgggg...?

**Comment by williewillus** [comment]
Because everyone has other things to do.
There are more polite ways to bump your pr besides saying that -.-

**Comment by ChunHin** [comment]
So how to bump my PR, since this is my first PR?

**Comment by Actuarius** [comment]
@williewillus added labels [Feature]

**Pull request closed without merging**